doc/HttpFiltering.md

```diff
@@ -0,0 +1,55 @@
+## HTTP Filtering
+
+With the release of sync-android version TBD a new HTTP  Filtering API was introduced.
+HTTP filtering allows the developer to modify the HTTP Requests and Responses during
+the replication process.
+
+In order to make changes to a HTTP request, you need to implement the interface
+`HTTPConnectionRequestFilter`, for HTTP Responses, you implement the interface
+`HTTPConnectionResponseFilter`. A Class may implement both interface without causing
+problems, an example of this is the `CookieFilter` class.
+
+In order to add an HttpFilter to a replication, you call the `addResponseFilters` or
+`addRequestFilters` on the `ReplicatorBuilder` class.
+
+For example, this is how you would add an instance of `CookieFilter` to a pull replication.
+
+```java
+import com.cloudant.sync.replication.Replicator
+import com.cloudant.sync.replication.ReplicatorBuilder
+import com.cloudant.sync.datastore.Datastore
+import com.cloudant.sync.http.CookieFilter
+
+
+Datastore ds =  manager.open("my_datastore");
+URI cloudant = new URI("https://username.cloudant.com");
+CookieFilter filter = new CookieFilter("username","password");
+
+
+
+Replicator replicator = ReplicatorBuilder.pull()
+							.from(cloudant)
```

Comment:
The formatting is odd here, with hard tabs. I think for legibility we should probably only indent these lines by one tab.